.github/workflows/build-check.yml

```diff
@@ -22,13 +22,13 @@ jobs:
           chmod +x gradlew
       - name: Gradle build jar
         run: |
-          ./gradlew --no-daemon clean jar
+          ./gradlew --no-daemon clean jar test
```

Comment:
Why not:
```
- name: Gradle test
  run: |
    ./gradlew --no-daemon test
```
The benefit is that tests get run in a separate step, making it easier to identify that it was the tests that failed, not anything else in the build.

It also might be beneficial to run the gradle daemon in this case.

Reply:
The documentation appears to apply if and only if we use the `actions/cache` action, which we currently do not. Additionally, I suspect there's a way to ensure the daemon gets killed regardless of job outcomes, but I'd need to look into it.

Reply:
I've removed the `--no-daemon` flag for this PR, but we may have to add it back in the future if we encounter issues.